gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/avro/AvroKeyCompactorOutputCommitter.java

```diff
@@ -0,0 +1,82 @@
+package gobblin.compaction.mapreduce.avro;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+import org.apache.hadoop.mapreduce.Counter;
+import org.apache.hadoop.mapreduce.StatusReporter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.output.FileOutputCommitter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Class used with {@link MRCompactorAvroKeyDedupJobRunner} to rename files as they
+ * are being committed. In addition to moving files from their working directory to
+ * the commit output directory, the files are named to include a timestamp and a
+ * count of how many records the file contains, in the format
+ * {recordCount}.{timestamp}.avro.
+ */
+public class AvroKeyCompactorOutputCommitter extends FileOutputCommitter {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AvroKeyCompactorOutputCommitter.class);
+
+  public AvroKeyCompactorOutputCommitter(Path output, TaskAttemptContext context)
+      throws IOException {
+    super(output, context);
+  }
+
+  /**
+   * Commits the task, moving files to their final committed location by delegating to
+   * {@link FileOutputCommitter} to perform the actual moving. First, renames the
+   * files to include the count of records contained within the file and a timestamp,
+   * in the form {recordCount}.{timestamp}.avro. Then, the files are moved to their
+   * committed location.
+   */
+  @Override
+  public void commitTask(TaskAttemptContext context) throws IOException {
+
+    Path workPath = getWorkPath();
+    FileSystem fs = workPath.getFileSystem(context.getConfiguration());
+
+    if (fs.exists(workPath)) {
+      long recordCount = getRecordCountCounter(context);
+      String fileName = recordCount + "." + System.currentTimeMillis() + ".avro";
```

Comment:
please add "part-r-" as a prefix.
Also, please add a random number as another component, in order to prevent the unlikely case that two files happen to have the same record counts and same timestamp.